packages/flutter_tools/test/general.shard/generate_localizations_test.dart

```diff
@@ -973,27 +894,104 @@ void main() {
     fail('Setting headerFile that does not exist should fail');
   });
 
-  testUsingContext('setting useDefferedLoading to null should fail', () {
-    _standardFlutterDirectoryL10nSetup(fs);
+  group('generateLocalizations', () {
+    testWithoutContext('forwards arguments correctly', () async {
+      _standardFlutterDirectoryL10nSetup(fs);
+      final LocalizationOptions options = LocalizationOptions(
+        header: 'HEADER',
+        arbDirectory: Uri.directory(defaultL10nPathString),
+        deferredLoading: true,
+        outputClass: 'Foo',
+        outputLocalizationsFile: Uri.file('bar', windows: false),
+        outputDirectory: Uri.directory(defaultL10nPathString, windows: false),
+        preferredSupportedLocales: <String>['es'],
+        templateArbFile: Uri.file(defaultTemplateArbFileName, windows: false),
+        untranslatedMessagesFile: Uri.file('untranslated', windows: false),
+        useSyntheticPackage: false,
+        areResourceAttributesRequired: true,
+        usesNullableGetter: false,
+      );
 
-    LocalizationsGenerator generator;
-    try {
-      generator = LocalizationsGenerator(fs);
-      generator.initialize(
-        inputPathString: defaultL10nPathString,
-        outputPathString: defaultL10nPathString,
-        templateArbFileName: defaultTemplateArbFileName,
-        outputFileString: defaultOutputFileString,
-        classNameString: defaultClassNameString,
-        headerString: '/// Sample header',
-        useDeferredLoading: null,
+      // Verify that values are correctly passed through the localizations target.
+      final LocalizationsGenerator generator = generateLocalizations(
+        fileSystem: fs,
+        options: options,
+        logger: BufferLogger.test(),
+        projectDir: fs.currentDirectory,
+        dependenciesDir: fs.currentDirectory,
       );
-    } on L10nException catch (e) {
-      expect(e.message, contains('useDeferredLoading argument cannot be null.'));
-      return;
-    }
 
-    fail('Setting useDefferedLoading to null should fail');
+      expect(generator.inputDirectory.path, '/lib/l10n/');
+      expect(generator.outputDirectory.path, '/lib/l10n/');
+      expect(generator.templateArbFile.path, '/lib/l10n/app_en.arb');
+      expect(generator.baseOutputFile.path, '/lib/l10n/bar');
```

Comment:
I believe these paths are failing the [Windows shard test](https://ci.chromium.org/ui/p/flutter/builders/try/Windows%20tool_tests_general/1138/overview) because `\` and `/`.

Reply:
Thanks, you were too quick, I was going to fix that before adding you to the review 😄